### DISCUSSION TITLE
RESTEasy Reactive - get all form data

### DISCUSSION THREAD

**Discussion by jrongs** [quarkusio/quarkus#25103] [Q&A] [answered]
with RESTEasy Reactive, is there an option to populate the form attributes map automatically for POST request? 

`@POST`
`@Path("/launch")`
`public String launch(HttpServerRequest request) {`
 `   MultiMap params = request.formAttributes() ;   //is always empty`
`}`

I understand that the map is empty due to the fact that request end handler is not yet called at this point,  what is the best approach to populate the form attributes before reaching this method?  I do know that we could use @RestForm annotations etc,  but lets assume we don't know what'll be posted here ahead of time.  any hint on this will be greatly appreciated, Thanks!

[a comment by an automated account is not shown]

**Comment by geoand** [comment, marked as answer]
So you essentially want a way to retrieve all form parameters, correct?

I wonder if we should expose something like we do for multipart. WDYT @stuartwdouglas @FroMage ?

**Reply by FroMage** [reply]
So, I've needed this in https://github.com/quarkusio/quarkus/issues/22444 for a long time too.

One solution that would work for both cases would be a `@RestReadForm` (name open to debate) which we can place on an endpoint class or method, or a filter (in my case) which would force reading the form body by adding the form filter, just as if we had form parameters. WDYT?

**Reply by geoand** [reply]
Seems reasonable to me

**Reply by naah69** [reply]
> Seems reasonable to me

is there any progress?
we need this too.

**Reply by FroMage** [reply]
In https://github.com/quarkusio/quarkus/issues/22444 we're adding `@WithFormRead` which you can put on your method to force reading the form values.

**Reply by Brutus5000** [reply]
Can someone explain what @WithFormRead changes? I tried it but `request.formAttributes()` still remains empty.
The only way I have right now is get the body as string and manually split by `&` and `=`

**Reply by jrongs** [reply]
@Brutus5000   take look at issue #22209, basically read from ResteasyReactiveRequestContext
